dabest/_dabest_object.py

```diff
@@ -479,7 +480,7 @@ def _check_errors(self, x, y, idx, experiment, experiment_label, x1_level):
 
             # Handling str type condition
             if is_str_condition_met:
-                if len(pd.unique(idx).tolist()) != 2:
+                if len(np.unique(idx).tolist()) != 2:
```

Comment:
Use length of the output instead of the output itself so can use np.unique() to avoid deprecation warning from pandas. 

np.unique() doesn't generate in occurrence order, might need returned indices to sort.